include/amber/shader_info.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2018 The Amber Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AMBER_SHADER_INFO_H_
+#define AMBER_SHADER_INFO_H_
+
+#include <string>
+#include <vector>
+
+namespace amber {
+
+enum class ShaderFormat : uint8_t {
+  kDefault = 0,
+  kText,
+  kGlsl,
+  kSpirvAsm,
+  kSpirvHex,
+};
+
+enum class ShaderType : uint8_t {
+  kCompute = 0,
+  kGeometry,
+  kFragment,
+  kVertex,
+  kTessellationControl,
+  kTessellationEvaluation,
+};
+
+struct ShaderInfo {
+  ShaderFormat format;
+  ShaderType type;
+  std::string shader_name;
+  std::string shader_data;
```

Comment:
Is the member name as clear as it could be?  Could use a comment.

A shader is the code and variables for a particular entry point in a SPIR-V module.   This field is really the source (from the vkscript or amberscript) for the SPIR-V module that contains the shader or shaders.

Suggest comment:

// Shader module contents, in the format specified by |format|